qiskit/algorithms/fidelities/fidelity_result.py

```diff
@@ -0,0 +1,32 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Fidelity result class
+"""
+
+from __future__ import annotations
+
+from typing import Any
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class FidelityResult:
+    """Result of Fidelity computation.
+
+    Args:
+        values (list[float]): List of fidelity values for each pair of input circuits.
+        metadata: Additional information on the fidelity calculations.
```

Comment:
If values needs/has the type here maybe we need it on metadata too.

Reply:
Are the typehints actually necessary in the docstring? I would think they're not since they are already on the variable

Reply:
I am not sure about that. I looked at the primitives in Terra which do both. I think its also possible to put the text along with the field rather than have it as Args. It needs to be checked in the resultant text which is why I used "maybe" in there. Things need to come out correct in the docs - it would be nice to avoid having to duplicate the typehints for sure.

Reply:
I have tried to follow the style used in the primitive's result classes: including the type in the doctoring and using more specific types. But maybe this is not necessary at this level. Honestly, I am happy to use `Sequence`/`Mapping` and not include the type in the doctoring (as done everywhere else in this PR). I will go for that option, if you have any issues with it, please comment again :)

Reply:
I would take a look at the output docs. My recollection is that if you specify Args these will show up to as args to the class constructor. In this case without types now which is not so good. The alternative is to not have Args and put the comments with the fields e.g. https://github.com/Qiskit/qiskit-terra/blob/bab9d4568334d572dd14ffef3b35567bb81fbc2c/qiskit/algorithms/optimizers/steppable_optimizer.py#L51-L73

Reply:
True, I have updated the class with comments in the fields, I think this looks clearer.